06_puppy/pfannerj/syncstore.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+// SyncStore is a sync.Map based in-memory implementation of PuppyStorer.
+type SyncStore struct {
+	sync.Map
+	currID uint32
+}
+
+// NewSyncStore creates a new in-memory store with sync map intialised.
+func NewSyncStore() *SyncStore {
+	return &SyncStore{}
+}
+
+// CreatePuppy adds a new puppy to the sync store.
+func (s *SyncStore) CreatePuppy(puppy *Puppy) (uint32, error) {
+	s.currID++
+	puppy.ID = s.currID
+	s.Store(puppy.ID, *puppy)
+	return puppy.ID, nil
+}
+
+// ReadPuppy gets a puppy from the sync store with the given ID.
+func (s *SyncStore) ReadPuppy(puppyID uint32) (*Puppy, error) {
+	if puppy, ok := s.Load(puppyID); ok {
+		if puppyOut, ok := puppy.(Puppy); ok {
```

Comment:
I'd force a panic if the cast goes wrong with `puppyOut := puppy.(Puppy)` 
(you get a wrong error message anyway with what you have).

Reply:
Fixed this as discussed...